File: C++/C++Basic/08.for-Loop-Exercise/04.clever-Lily.cpp
#include <iostream>

using namespace std;

int main()
{
    int ages;
    double washingMachine;
    int priceToy;

    cin >> ages;
    cin >> washingMachine;
    cin >> priceToy;

    double price = 10.0;
    int countToys = 0;
    int totalSum = 0;
    int a = 4;

    for (int age = 1; age <= ages; age++)
    {
        if (age % 2 == 0)
        {
            totalSum += price;
            price += 10;
            totalSum -= 1;

        }
        else
        {
           countToys ++;
        }
    }
    
    totalSum += countToys * priceToy;

    cout.setf(ios::fixed);
    cout.precision(2);

    if (totalSum >= washingMachine)
    {
        cout <<"Yes! " << abs (totalSum-washingMachine) << endl;
    }
    else
    {
        cout << "No! " << abs(totalSum - washingMachine) << endl;
    }

    return 0;
}

File: Python/Basic/08.For-Loop-Exersice/04.clever_lily.py
ages = int(input())
washing_machine_price = float(input())
price_toys = int(input())

money_save = 10
total_sum = 0
count_toys = 0


for age in range(1, ages+1):

    if (age % 2 == 0):
        total_sum  += money_save
        money_save += 10
        total_sum -= 1
    else:
        count_toys += 1

total_sum += count_toys * price_toys

if total_sum >= washing_machine_price:
    print(f"Yes! {abs(total_sum - washing_machine_price):.2f}")
else:
    print(f"No! {abs(total_sum - washing_machine_price):.2f}")